src/megaclient.cpp

```diff
@@ -2766,18 +2769,17 @@ void MegaClient::exec()
                             // filesystem item is notified or initiate a full rescan if there has been
                             // an event notification failure (or event notification is unavailable)
                             bool scanfailed = false;
-                            for (it = syncs.begin(); it != syncs.end(); it++)
+                            bool noneSkipped = true;
+                            for (Sync* sync : syncs)
                             {
-                                Sync* sync = *it;
-
                                 totalnodes += sync->localnodes[FILENODE] + sync->localnodes[FOLDERNODE];
 
                                 if (sync->state == SYNC_ACTIVE || sync->state == SYNC_INITIALSCAN)
                                 {
                                     if (sync->dirnotify->notifyq[DirNotify::DIREVENTS].size()
                                      || sync->dirnotify->notifyq[DirNotify::RETRY].size())
                                     {
-                                        break;
+                                        noneSkipped = false;
```

Comment:
Yes I think so too - the break here meant that we treated syncs differently depending on the order that they had in the list of syncs, which seems wrong, sync behaviour should be independent of that, otherwise we're making the system more difficult to diagnose, and possibly skipping some needed work on those subsequent syncs.   I didn't see anything in this loop that would warrant skipping processing subsequent syncs.   Good spotting, by the way.